### PR TITLE
Fix phpstan issues

### PR DESCRIPTION
**Pull request by dingo-d**
I've had to exclude some checks as they are false positive. Most of the config calls are false-positive because they depend on the `Config_Data` interface, which dictates that those methods have to exist - but they will be implemented in the theme or a plugin, and there are no concrete implementations in the libs, so it says it's an error when it's not.

Walker has core inconsistencies (https://core.trac.wordpress.org/ticket/50261) that are throwing false-positives.

I've made some changes so it would be great if we test this before merging 😬 

Ideally, we'd have automated tests but I need to work on those...

### CODE REVIEW

**Comment by iruzevic** [comment]
this looks good, please merge latest develop branch to your branch to resolve conflicts

**Comment by szepeviktor** [comment]
> Call to an undefined method object::get_config().

Those are easy to fix!

If you merge this one I'll send a PR right away.

File: src/class-main.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ private function get_service_classes_prepared_array() : array {
     $output = [];
 
     foreach ( $this->get_service_classes() as $class => $dependencies ) {
-      if ( is_array( $dependencies ) ) { /** @phpstan-ignore-line */
+      if ( is_array( $dependencies ) ) { /** @phpstan-ignore-line phpcs:ignore */
```
Comment (szepeviktor) [inline src/class-main.php:147]: Just set the return type of `get_service_classes`

`array<class-string, string|string[]>`

Comment (dingo-d) [reply src/class-main.php]: I had them set to just `array<string>` which caused errors. Thanks for the tip 👍

**Comment by dingo-d** [comment]
@szepeviktor You can make the PR if you want with further fixes 👍

**Comment by szepeviktor** [comment]
🏎️ Starting **now**...

**Pull request merged**